includes/Data/ContentBlocksResolver.php

```diff
@@ -242,6 +243,33 @@ private static function populate_post_content_inner_blocks( array $block ): arra
 		return $block;
 	}
 
+	/**
+	 * Populates the innerBlocks of this block with navigation item blocks from the referenced navigation post.
+	 *
+	 * @param array<string,mixed> $block The block to populate.
+	 *
+	 * @return array<string,mixed> The populated block.
+	 */
+	private static function populate_navigation_blocks( array $block ): array {
+		if ( 'core/navigation' !== $block['blockName'] || ! isset( $block['attrs']['ref'] ) ) {
+			return $block;
+		}
+
+		$navigation_post = get_post( $block['attrs']['ref'] );
+
+		if ( ! $navigation_post || 'publish' !== $navigation_post->post_status ) {
```

Comment:
Can we check if the `$navigation_post instanceof WP_Post` here instead of `! $navigation`?

Reply:
I don't think this is necessary since this is how Gutenberg checks it as well

https://github.com/WordPress/gutenberg/blob/trunk/packages/block-library/src/navigation/index.php#L230-L241

Reply:
With all due respect to the Gutenberg code sample here, we are using the `$navigation_post` as an instance of a WP_Post on next lines. If we decide to tighten things up with phpstan in future that might be an issue as.